net/frr/src/opnsense/service/templates/OPNsense/Quagga/bgpd.conf

```diff
@@ -219,8 +218,10 @@ route-map {{ routemap.name }} {{ routemap.action }} {{ routemap.id }}
 {%             endif %}
 {%           endfor %}
 {%         endif %}
-{%         if routemap.set|default("") != '' and routemap.match2|default("") != '' %}
- set {{ routemap.set }}
+{%         if routemap.set|default("") != '' %}
```

Comment:
If I'm understanding the documentation correctly the valid options to set are these https://docs.frrouting.org/en/latest/routemap.html#route-map-set-command

At a first glance using comma's to separate these items should be safe in terms that their not being used for the input, but from a validation perspective this still is an awful construct and may lead to code injections at some point. We could consider changing the input to a text area, but also offer the set it's own field type (https://docs.opnsense.org/development/frontend/models_customfields.html) so we validate each line according to the documentation. It's more work, but likely simplifies input as well.

Reply:
I'll take a look into that.
I am fairly new to this whole topic so bare with me.

Reply:
no problem, my time is a bit limited, if it helps I can create a separate branch with the boilerplate code so you can extend it further from there (and we'll just squash the changes to master when the topic is resolved).  I don't expect it will cost much time on my end and might help explaining what I mean. The list of valid "set" options looks limited enough if I do understand the documentation.